feat: Adiciona rota GET "climate"

Ivan-Duarte opened the pull request:
1. Essa rota é responsável por garantir que os dados inseridos no formulário de pesquisa no frontend sejam tratados no backend e formatados em um JSON personalizado seguindo o seguinte padrão:

    Retorna:
```
        dict: Um JSON formatado contendo os dados no seguinte formato:
              {
                  "message": "Resposta API - Clima",
                  "status": "sucesso",
                  "data": [
                      {
                          "day": "YYYYMMDD",
                          "temperature": valor_da_temperatura,
                          "precipitation": valor_da_precipitação
                      }
                  ]
              }
```

2. Além da rota, foram adicionadas alterações no README com comandos atualizados de inicialização do BackEnd, bem como função de test para o método get_temporal_data, presente no arquivo "climatology.py".
![image](https://github.com/user-attachments/assets/9ba7c38f-fffa-4b7f-880d-96f0a55c0919)

4. Restruturação das pastas do BackEnd:
![image](https://github.com/user-attachments/assets/387391cb-c05e-4636-9709-a2f5244c4c88)

5. Adição do parametro PRECTOTCORR a rota GET "climate.py"
![image](https://github.com/user-attachments/assets/2f0d2d4c-6ab7-4793-a622-4f73f7965e08)

Ivan-Duarte committed:
docs: Atualiza a documentação do ReadME com comandos atuais para inicialização do projeto.

Ivan-Duarte committed:
feat: Adiciona Rota GET para dados de Temperatura

Ivan-Duarte committed:
docs: Atualiza a documentação do ReadME com comandos atuais para inicialização do projeto.

KhovetS2 commented:
Favor, faz a alteração de remover o parameters e colocar ele já constante no código e adiciona também o  request da pluviosidade a tag para api da nasa é essa PRECTOTCORR e adiciona a resposta a pluviosidade.

Ivan-Duarte committed:
refactor:Adiciona parametro PRECTOTCORR a rota GET

+ Modifica a função 'formatted_response.py' para enviar JSON formatado
  agora com novo parametro.
+ Modifica a rota 'climate.py' para receber somente parametros de
  latitude, longitude, dada inicio e dada fim.
